### PR TITLE
feat: Support disabling storage autoscaling

### LINKED ISSUE #14: Auto scaling target/policy creation failing in il-central-1
## Description

In regions like il-central-1(Tel Aviv, Israel), we have MSK but it does not support autoscaling groups yet. In such cases when we try to create MSK using this module, it breaks with a "Not Authorized to perform this action" exception. 

If your request is for a new feature, please use the `Feature request` template.

- [Yes ] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: Latest

- Terraform version: 1.4.6

- Provider version(s): 5.16.2 (Hashicorp/aws)

## Reproduction Code [Required]

 Create AWS MSK in il-central-1 using this terraform module.

## Expected behavior
Should work. (Making autoscaling targets/policies optional will do the job)

## Actual behavior
Breaking.

### PR DESCRIPTION
**Pull request by nishsak96**
…resources

## Description
Added boolean option for creation of appautoscaling target and policy

## Motivation and Context

 MSK creation failing in regions without support for autoscaling resources

Resolves #14

## Breaking Changes
No.

## How Has This Been Tested?
- [ No ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ No ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ No ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -192,7 +192,7 @@ resource "aws_cloudwatch_log_group" "this" {
 ################################################################################
 
 resource "aws_appautoscaling_target" "this" {
-  count = var.create ? 1 : 0
+  count = var.create && var.create_appautoscaling_target ? 1 : 0
```
Comment (bryantbiggs) [inline main.tf:195]: lets just create one variable that is used to enable/disable autoscaling
```suggestion
  count = var.create && var.enable_storage_autoscaling ? 1 : 0
```

File: main.tf [hunk 2]
```diff
@@ -203,7 +203,7 @@ resource "aws_appautoscaling_target" "this" {
 }
 
 resource "aws_appautoscaling_policy" "this" {
-  count = var.create ? 1 : 0
+  count = var.create && var.create_appautoscaling_policy && var.create_appautoscaling_target ? 1 : 0
```
Comment (bryantbiggs) [inline main.tf:206]:
```suggestion
  count = var.create && var.enable_storage_autoscaling ? 1 : 0
```

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 2.3.0 :tada:

**Comment by nishsak96** [comment]
Thanks @bryantbiggs and @antonbabenko.

**Pull request merged**